### PR TITLE
EAM: Fix use of real(r8) rather than integer in index arrays.

### PR DESCRIPTION
**Pull request by ambrad**
Fixes index type to be integer, as intended.

[BFB]

### CODE REVIEW

**Comment by ambrad** [comment]
@grnydawn when you get a chance, would you test this branch in the environment where you're seeing the issues? This PR cleans up the warnings on pm-cpu.

One easy way to do this with whatever your current working branch is:
```
wget https://patch-diff.githubusercontent.com/raw/E3SM-Project/E3SM/pull/5805.patch
git am 5805.patch
```

**Comment by ambrad** [comment]
e3sm_developer passes on Chrysalis.

File: components/eam/src/physics/cam/zm_microphysics.F90 [hunk 1]
```diff
@@ -2713,7 +2713,7 @@ subroutine zm_mphy(su,    qu,   mu,   du,   eu,    cmel,  cmei,  zf,   pm,   te,
                  qnitend(i,k) = (prai(k)+psacws(k)+prci(k))+pracs(k)-psacr(k)+piacrs(k)   &
                                  +pracis(k)
                  
-                 qgtend(i,k) = (pracg(k)+psacwg(k)+pgsacw(k)+pgracs(k)+ &
+                 qgtend(i,k) = (pracg(k)+psacwg(k)+pgsacw(k)+pgracs(k) &
```
[the remark this answers is not in the document]
Comment (ambrad) [reply components/eam/src/physics/cam/zm_microphysics.F90:2716]: Yes, some compilers warn about it. Note the next line. One sees that there was `x + +y`, and `+` is not a unary operator in strict Fortran.

**Review by grnydawn — APPROVED** [review]
This fix resolved the issue that I got from using the Cray compiler on Crusher.

**Review by sarats — APPROVED** [review]

**Pull request merged**